lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module

```diff
@@ -696,3 +696,41 @@ function dosomething_helpers_get_campaign_status($start_date, $end_date) {
     return 'closed';
   }
 }
+
+/**
+ * Gets the current run for a given campaign nid and user.
+ *
+ * @param int $nid
+ * @param object $user
+ *  (optional) Will default to global $user if not supplied
+ *
+ * @return
+ *  Returns the campaigns node in the correct language or false.
+*/
+function dosomething_helpers_get_campaign_run($nid, $user = NULL) {
+  // Setup variables
+  $node = node_load($nid);
+  if (!isset($user)) {
+    global $user;
+  }
+
+  // First check if we have a run in the user lang
+  $user_language = dosomething_global_get_language($user, $node);
+  $run_nid = dosomething_helpers_extract_field_data($node->field_current_run, $user_language);
+
+  // If not try the nodes source language
+  if (!isset($run)) {
```

Comment:
How is this checking if `$run` is set, if it doesn't get assign beforehand? It will always fail won't it?